spec/acceptance/init_spec.rb

```diff
@@ -20,8 +20,27 @@ class { 'bolt': use_release_package => false }
     end
   end
 
-  describe command('yum clean all --verbose; rm -rf /etc/yum.repos.d/puppet-tools-release.repo; yum erase --assumeyes puppet-bolt puppet-tools-release') do
-    its(:exit_status) { is_expected.to eq 0 }
+  describe 'with use_release_package=false & ensure=absent' do
+    it_behaves_like 'an idempotent resource' do
+      let(:manifest) do
+        <<-PUPPET
+        class { 'bolt':
+          use_release_package => false,
+          version => 'absent',
+        }
+        PUPPET
+      end
+    end
+
+    describe 'packages' do
+      it { expect(package('puppet-bolt')).not_to be_installed }
+      it { expect(package('puppet-tools-release')).not_to be_installed }
+    end
```

Comment:
This may also work:
```suggestion
    it_behaves_like 'an idempotent resource' do
      let(:manifest) do
        <<-PUPPET
        class { 'bolt':
          use_release_package => false,
          version => 'absent',
        }
        PUPPET
      end

      it { expect(package('puppet-bolt')).not_to be_installed }
      it { expect(package('puppet-tools-release')).not_to be_installed }
    end
```

Reply:
won't work, the `it_behaves_like` will be executed after the package checks.